### DISCUSSION TITLE
What are the biggest learning curves in Github?

### DISCUSSION THREAD

**Discussion by novium2** [community/community#138591] [Discover: GitHub Best Practices]
Where did you struggle most as a beginner when you first started using Github? What do you see people struggling with most? For me, it's a tie between dialing in all the notifications and getting accustomed to working with forks, branches, and pull requests and fitting them into established workflows. What about you? What do you think the biggest learning curves are, and what helped you through them?

**Comment by jovanovic-djo** [comment]
When I first started using Github, struggle was understanding the flow of branches and pull requests, figuring out when to branch off, make commits, and how to properly merge everything back in without causing conflicts took some time. 

You can try out this, I used it for introduction purposes in the beginning: https://learngitbranching.js.org/ 

What really helped me through it was taking the time to play around in smaller personal projects, don't be afraid to make mistakes in the beginning, its the part of the process. 

And suddenly, it will all click.

**Reply by novium2** [reply]
@jovanovic-djo did you find a good rule of thumb for when to use branches?

**Reply by jovanovic-djo** [reply]
Yes, for the beginning, whenever I want to experiment with code or develop new feature, I checkout to new branch, I usually call it "development". Then, when I am done, i merge changes to the main.

Later on, through the career, you will be introduced to other, more advanced practices.

**Reply by novium2** [reply]
So if I'm hearing you right, @jovanovic-djo , anything that's maybe a bit more substantial - like experimenting, or developing a new feature - is a good option for a branch, whereas maybe something like correcting a typo or some other minor problem is a better fit for just a plain pull request?

**Reply by jovanovic-djo** [reply]
I would definitely not use new branch for typo, but for minor problem, there is bugfix branch.
Here is documentation: https://gist.github.com/jpolete/aa31b9043e8e90f8a47c7738669555fa
Be consistent about reading documentation, experimenting and exploring, I am sure it all make sense through few weeks.

Have fun.

**Comment by Amanacer** [comment]
Mastering GitHub involves steep learning curves, particularly in understanding version control with Git, navigating its command-line interface, and mastering branching and merging strategies. Adapting to collaborative workflows through pull requests and issues, along with automating tasks with GitHub Actions, adds to the complexity. Security, repository management, and troubleshooting also pose significant challenges. Overcoming these hurdles requires practice, patience, and leveraging available resources.
